pkg/cmd/cli/cmd/create/clusterquota.go

```diff
@@ -0,0 +1,147 @@
+package create
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	quotaapi "github.com/openshift/origin/pkg/quota/api"
+)
+
+const (
+	ClusterQuotaRecommendedName = "clusterresourcequota"
+
+	clusterQuotaLong = `
+Create a cluster resource quota that controls certain resources.
+
+Cluster resource quota objects defined quota restrictions that span multiple projects based on label selectors.`
+
+	clusterQuotaExample = `  # Create an cluster resource quota limited to 10 pods
+  %[1]s limit-bob --project-selector=openshift.io/requester=user-bob --hard=pods=10`
+)
+
+type CreateClusterQuotaOptions struct {
+	ClusterQuota *quotaapi.ClusterResourceQuota
+	Client       client.ClusterResourceQuotasInterface
+
+	Mapper       meta.RESTMapper
+	OutputFormat string
+	Out          io.Writer
+	Printer      ObjectPrinter
+}
+
+// NewCmdCreateServiceAccount is a macro command to create a new service account
+func NewCmdCreateClusterQuota(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	o := &CreateClusterQuotaOptions{Out: out}
+
+	cmd := &cobra.Command{
+		Use:     name + " NAME --project-selector=key=value [--hard=RESOURCE=QUANTITY]...",
+		Short:   "Create cluster resource quota resource.",
+		Long:    clusterQuotaLong,
+		Example: fmt.Sprintf(clusterQuotaExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(cmd, f, args))
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Run())
+		},
+		Aliases: []string{"clusterquota"},
```

Comment:
> I would have this as the name and "clusterresourcequota" as a suggestion but that's just me conforming to kubectl conventions of no aliases. Nevermind that suggestions don't currently work for nested commands.

We've done this for short names before.  See `oc create deploymentconfig` and `oc create dc`.  Since the resource shortname is unambiguous I think this is more userfriendly.

@fabianofranz